playground/src/main.tsx

```diff
@@ -70,6 +75,12 @@ function App(props: { katas: Kata[]; linkedCode?: string }) {
   const [compiler, setCompiler] = useState(() =>
     createCompiler(setCompilerState),
   );
+
+  const [compiler_worker_factory] = useState(() => {
```

Comment:
`createCompiler` on line 66 looks like it might be suitable, and is already created as state on line 75 here. Is that reusable, or not appropriate for some reason?

Reply:
It isn't reusable. The qir compiler worker is expected to fail a lot and get runtime errors.